addons/web_editor/static/src/js/wysiwyg/widgets/link.js

```diff
@@ -183,8 +192,15 @@ const Link = Widget.extend({
     applyLinkToDom: function (data) {
         // Some mass mailing template use <a class="btn btn-link"> instead of just a simple <a>.
         // And we need to keep the classes because the a.btn.btn-link have some special css rules.
-        if (!data.classes.includes('btn') && this.data.iniClassName.includes("btn-link")) {
-            data.classes += " btn btn-link";
+        // Same thing for the "btn-success" class, this class cannot be added
+        // by the options but we still have to ensure that it is not removed if
+        // it exists in a template (e.g. "Newsletter Block" snippet).
+        if (!data.classes.split(' ').includes('btn')) {
+            for (const linkClass of this.toleratedClasses) {
+                if (this.data.iniClassNames && this.data.iniClassNames.split(' ').includes(linkClass)) {
+                    data.classes += " btn " + linkClass;
+                }
```

Comment:
Same here `this.data.iniClassNames` will always make your test fail and the `'btn-'` class will never be added.